### PR TITLE
Introduce compact header

### PR DESCRIPTION
**Pull request by corny**

### CODE REVIEW

File: kmod/fastd.c [hunk 1]
```diff
@@ -167,6 +169,7 @@ struct fastd_softc {
 	volatile u_int		refcnt;		/* reference counter */
 	char			pubkey[FASTD_PUBKEY_SIZE]; /* public key of the peer */
 	uint32_t		flags;
+	int					use_compact_header; /* use compact fastd header for data. */
```
Comment (corny) [inline kmod/fastd.c]: Please fix the indentation and use char.

File: kmod/fastd.h [hunk 2]
```diff
@@ -19,6 +19,8 @@ typedef union {
 
 #define FASTD_HDR_HANDSHAKE	0x01
 #define FASTD_HDR_DATA		0x02
+#define FASTD_HDR_IP4PACKET		0x40
+#define FASTD_HDR_IP6PACKET		0x60
```
Comment (corny) [inline kmod/fastd.h]: Please fix the indentation.

File: kmod/fastd.c [hunk 3]
```diff
@@ -794,6 +812,13 @@ fastd_rcv_udp_packet(struct mbuf *m, int offset, struct inpcb *inpcb,
 	m_copydata(m, offset, 1, (caddr_t) &msg_type);
 	rm_rlock(&fastd_lock, &tracker);
 
+	/* Take the higher nibble as the command value if the value is bigger
+	   than 0xf.  This is part of the compact header fastd extension.
+	*/
+	if (msg_type > 0xf) {
```
Comment (dmke) [inline kmod/fastd.c]: Please format nibbles in whole-byte-notation (`0x0f`).

Comment (0x5a17ed) [reply kmod/fastd.c]: done

**Pull request merged**